### PR TITLE
WIP: Users list is not alphabetically ordered on Org-> User page

### LINKED ISSUE #22371: Users list on org->User page is case sensitive
**What happened**:

Usernames/emails starting with Upper case letter are listed at the top followed by usernames/emails starting with lower case letters 

http://localhost:3000/org/users

image

**What you expected to happen**:

User list should be case insensitive 
image

**How to reproduce it (as minimally and precisely as possible)**:
Create users with upper and lower case letters

**Anything else we need to know?**:

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by vikkyomkar**
What this PR does / why we need it:
Users are listed in alphabetical order by their usernames on the Org Useer page.

Which issue(s) this PR fixes:

Fixes #22371

### CODE REVIEW

File: pkg/services/sqlstore/org_users.go [hunk 1]
```diff
@@ -129,7 +129,8 @@ func GetOrgUsers(query *m.GetOrgUsersQuery) error {
 		"org_user.role",
 		"user.last_seen_at",
 	)
-	sess.Asc("user.email", "user.login")
+	sess.OrderBy("upper(user.login), upper(user.email)")
+	//sess.Asc("user.email", "user.login")
```
Comment (aknuds1) [inline pkg/services/sqlstore/org_users.go]:
```suggestion
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]

File: pkg/services/sqlstore/org_users.go [hunk 2]
```diff
@@ -129,7 +129,8 @@ func GetOrgUsers(query *m.GetOrgUsersQuery) error {
 		"org_user.role",
 		"user.last_seen_at",
 	)
-	sess.Asc("user.email", "user.login")
+	sess.OrderBy("upper(user.login), upper(user.email)")
```
Comment (aknuds1) [inline pkg/services/sqlstore/org_users.go]: I thought the following ordering was agreed? 
```
sess.OrderBy("LOWER(u.login),u.login,LOWER(u.email),u.email")
```

**Comment by aknuds1** [comment]
Postgres integration tests are failing, any idea why?

**Comment by vikkyomkar** [comment]
> Postgres integration tests are failing, any idea why?

Its failing if `OrderBy` is used, it seems
Trying to understand more about this `Postgres integration tests` failure

**Comment by aknuds1** [comment]
The tests are still failing.

**Comment by aknuds1** [comment]
I synced your branch with latest Grafana master, since we've decided to have another look at it.

**Comment by marefr** [comment]
@aknuds1 I'm a bit worried using `lower()` will not be able to use any indices in database. Think the proper solution to this would be to always store/treat username and email in lowercase. We have another issue regarding that, https://github.com/grafana/grafana/issues/12549 but it's a bit tricky.

**Comment by aknuds1** [comment]
Thanks for the feedback @marefr - I haven't been able to get around to looking at this PR yet beyond syncing it with master. So what is your take wrt. this PR, is it infeasible or can we merge a modified version of it?

**Comment by aknuds1** [comment]
I'm truly sorry, but we're going to have to close this PR. The reason is that after analyzing the problem, we've come to the conclusion that ordering by login and login case insensitive has too much of a negative impact on query performance.

We think that fixing this issue will have to wait until we have been able to resolve the more fundamental issue of making usernames case insensitive.

**Pull request closed without merging**